lms/djangoapps/experiments/flags.py

```diff
@@ -128,6 +135,22 @@ def get_bucket(self, course_key=None, track=True):
             # We need username for stable bucketing and id for tracking, so just skip anonymous (not-logged-in) users
             return 0
 
+        if not is_masquerading(request.user, course_key):
```

Comment:
As talked about a bit in slack, I'd like it if we could avoid this first check/code-path. Can we just setup at the top of whatever endpoint isn't doing the setup now?  Doing it down deep in the stack feels weird to me.

Reply:
I can leave this out.